src/caffe/solvers/adam_solver.cu

```diff
@@ -6,12 +6,15 @@ namespace caffe {
 template <typename Dtype>
 __global__ void AdamUpdate(int N, int t, Dtype* g, Dtype* m, Dtype* v,
                            Dtype beta1, Dtype beta2, Dtype eps_hat, Dtype corrected_local_rate,
-                           bool amsgrad, bool rectified) {
+                           bool amsgrad, bool rectified, bool adabelief) {
   CUDA_KERNEL_LOOP(i, N) {
     float gi = g[i];
     float mi = m[i] = m[i]*beta1 + gi*(1-beta1);
     float vi_old = v[i];
-    float vi = v[i] = v[i]*beta2 + gi*gi*(1-beta2);
+    float vi;
+    if (adabelief)
+      gi -= mi;
+    vi = v[i] = v[i]*beta2 + gi*gi*(1-beta2);
```

Comment:
define `float vi` here instead of uninitialized above ?

Reply:
ok done